### DISCUSSION TITLE
Could not find required `intl` object

### DISCUSSION THREAD

**Discussion by kark** [commercetools/ui-kit#2182] [Q&A] [answered]
When using some of the uikit components I'm getting an error regarding a missing `intl` object.

```
Uncaught Error: [React Intl] Could not find required intl object.
<IntlProvider> needs to exist in the component ancestry.
```

How should I fix this?

---

Refs:
* #2158
* #2167

**Comment by kark** [comment, marked as answer]
Hi 👋 , 

This issue usually means that there are conflicting versions of `react-intl` installed in your repository. These kinds of dependency issues unfortunately occur from time to time.

What usually helps (from our experience) in verifying that this is the case is checking the installed versions with `yarn why react-intl`. This should show you if versions of `react-intl` installed in the repository are the same or there are conflicting versions.

If there are conflicting versions you can attempt to do the following steps:
* Force Yarn to install the dependencies to their latest version range with `yarn upgrade react-intl` (with Yarn v1) or `yarn up -R react-intl` (with Yarn v3).
* Use [Yarn `resolutions`](https://yarnpkg.com/configuration/manifest#resolutions) to force the resolution of a specific version.
* If the issue still persists, try removing `node_modules` and `yarn.lock` and re-install all dependencies.

**Reply by CarlosCortizasCT** [reply]
Perhaps we could also recall here that, in the context of a developing a Custom Application, `IntlProvider` is used within `ApplicationShell` component which users commonly use in their top component's tree so this problem does not come from using that provider incorrectly (users are not expected to actually use it in their code most of the time) but from the dependencies versions conflict.
